sql/signal_handler.cc

```diff
@@ -58,49 +58,70 @@ extern const char *optimizer_switch_names[];
 
 static inline void output_core_info()
 {
-  /* proc is optional on some BSDs so it can't hurt to look */
-#if defined(HAVE_READLINK) && !defined(__APPLE__) && !defined(__FreeBSD__)
+#ifndef __WIN__
   char buff[PATH_MAX];
   ssize_t len;
   int fd;
+  if ((fd= open("/etc/os-release", O_RDONLY)) >= 0)
+  {
+    len= read(fd, buff, sizeof(buff));
+    close(fd);
```

Comment:
I would add 
```
if (close(fd) < 0)
{
  error
}
```
only if read is successfull.

Reply:
we only close a file if a read is successful? nope - we close a file if it opened successfully, read errors will be handled. Like above, didn't want to put errors on information gathering to confuse with the original signal causing error.